jetty-http/src/main/java/org/eclipse/jetty/http/HttpURI.java

```diff
@@ -610,7 +610,7 @@ public String getFragment()
     /* ------------------------------------------------------------ */
     public void decodeQueryTo(MultiMap<String> parameters)
     {
-        if (_query==_fragment)
+        if (_query!=null && _query.equals(_fragment))
```

Comment:
Good catch.  We actually don't need to comparison with fragment.  We just need the not null test now